### ISSUE TITLE
TextSentiment .setOpinionMining not returning opinion mining results

### ISSUE THREAD

**Issue by kenshinee** [microsoft/SynapseML#1577]
**Describe the bug**

Using .setOpinionMining(True) in Pyspark does not return opinion mining results on Azure Synapse Analytics

**To Reproduce**

1. Using Pyspark, create the object e.g. (TextSentiment().setOpinionMining(True)....etc.
2. Run the results with a loaded dataframe
3. Results is indifferent from not setting the Opinion Mining

**Expected behavior**

Resulted JSON to include opinion mining results .i.e. "targets" and "assessments".

Also tested comparable lib (azure.ai) and Rest API calls (v3 and v22.05.01). All 3 are returning opinion mining results when the opinionMining flag/key is included.

**Info (please complete the following information):**
 - SynapseML Version: [e.g. v0.17]
 - Spark Version [e.g. 2.4.3]
 - Spark Platform [e.g. Databricks]

** Stacktrace**
```
Please post the stacktrace here if applicable
```

If the bug pertains to a specific feature please tag the appropriate [CODEOWNER](https://github.com/Microsoft/SynapseML/blob/master/CODEOWNERS) for better visibility

**Additional context**

AB#1890098

**Comment by kenshinee** [comment]
**Test code in Azure Synapse Analytics**

`columns = StructType([StructField('Id', StringType(), False),
                       StructField('Comments', StringType(), False)])

vals = [(1, "Food is great"),(2, "Service is poor but the restaurant is at a good location")]                       
firstDF = spark.createDataFrame(vals, schema = columns)                        
secondDF = spark.createDataFrame(vals, schema = columns)                        

AnalyticsTS = (TextSentiment()                        
    .setLinkedService(CogSvcLinked)        
    .setOpinionMining(True)
    .setLanguage("en")
    .setOutputCol("Prediction")
    .setErrorCol("Error")        
    .setTextCol("Comments"))

AnalyticsTS2 = (TextSentiment()                        
    .setLinkedService(CogSvcLinked)        
    .setLanguage("en")
    .setOutputCol("Prediction")
    .setErrorCol("Error")        
    .setTextCol("Comments"))

ResultsTSDF = AnalyticsTS.transform(firstDF)
ResultsTSDF2 = AnalyticsTS2.transform(secondDF)

print ("With Opinion Mining")
ResultsTSDF.show(truncate=False)

print ("Without Opinion Mining")
ResultsTSDF2.show(truncate=False)`

**Results are the same**

**With Opinion Mining**
+---+--------------------------------------------------------+-----+-------------------------------------------------------------------------------------------------------------------------------------------------+
|Id |Comments                                                |Error|Prediction                                                                                                                                       |
+---+--------------------------------------------------------+-----+-------------------------------------------------------------------------------------------------------------------------------------------------+
|1  |Food is great                                           |null |[{positive, null, {1.0, 0.0, 0.0}, [{Food is great, positive, {1.0, 0.0, 0.0}, 0, 13}], [], null}]                                               |
|2  |Service is poor but the restaurant is at a good location|null |[{negative, null, {0.29, 0.01, 0.7}, [{Service is poor but the restaurant is at a good location, negative, {0.29, 0.01, 0.7}, 0, 56}], [], null}]|
+---+--------------------------------------------------------+-----+-------------------------------------------------------------------------------------------------------------------------------------------------+

**Without Opinion Mining**
+---+--------------------------------------------------------+-----+-------------------------------------------------------------------------------------------------------------------------------------------------+
|Id |Comments                                                |Error|Prediction                                                                                                                                       |
+---+--------------------------------------------------------+-----+-------------------------------------------------------------------------------------------------------------------------------------------------+
|1  |Food is great                                           |null |[{positive, null, {1.0, 0.0, 0.0}, [{Food is great, positive, {1.0, 0.0, 0.0}, 0, 13}], [], null}]                                               |
|2  |Service is poor but the restaurant is at a good location|null |[{negative, null, {0.29, 0.01, 0.7}, [{Service is poor but the restaurant is at a good location, negative, {0.29, 0.01, 0.7}, 0, 56}], [], null}]|
+---+--------------------------------------------------------+-----+-------------------------------------------------------------------------------------------------------------------------------------------------+



**Test using Azure.AI with Opinion Mining**
i.e. analyzed_reviews = client.analyze_sentiment(reviews, show_opinion_mining=True)

**Results:**
`[{
		'id': '0',
		'sentiment': 'positive',
		'warnings': [],
		'statistics': None,
		'confidence_scores': SentimentConfidenceScores(positive = 1.0, neutral = 0.0, negative = 0.0),
		'sentences': [SentenceSentiment(text = Food is great, sentiment = positive, confidence_scores = SentimentConfidenceScores(positive = 1.0, neutral = 0.0, negative = 0.0), length = 13, offset = 0, mined_opinions = [MinedOpinion(target = TargetSentiment(text = Food, sentiment = positive, confidence_scores = SentimentConfidenceScores(positive = 1.0, neutral = 0.0, negative = 0.0), length = 4, offset = 0), assessments = [AssessmentSentiment(text = great, sentiment = positive, confidence_scores = SentimentConfidenceScores(positive = 1.0, neutral = 0.0, negative = 0.0), length = 5, offset = 8, is_negated = False)])])],
		'is_error': False
	},
	{
		'id': '1',
		'sentiment': 'negative',
		'warnings': [],
		'statistics': None,
		'confidence_scores': SentimentConfidenceScores(positive = 0.29, neutral = 0.01, negative = 0.7),
		'sentences': [SentenceSentiment(text = Service is poor but the restaurant is at a good location, sentiment = negative, confidence_scores = SentimentConfidenceScores(positive = 0.29, neutral = 0.01, negative = 0.7), length = 56, offset = 0, mined_opinions = [MinedOpinion(target = TargetSentiment(text = Service, sentiment = negative, confidence_scores = SentimentConfidenceScores(positive = 0.01, neutral = 0.0, negative = 0.99), length = 7, offset = 0), assessments = [AssessmentSentiment(text = poor, sentiment = negative, confidence_scores = SentimentConfidenceScores(positive = 0.01, neutral = 0.0, negative = 0.99), length = 4, offset = 11, is_negated = False)]), MinedOpinion(target = TargetSentiment(text = location, sentiment = positive, confidence_scores = SentimentConfidenceScores(positive = 1.0, neutral = 0.0, negative = 0.0), length = 8, offset = 48), assessments = [AssessmentSentiment(text = good, sentiment = positive, confidence_scores = SentimentConfidenceScores(positive = 1.0, neutral = 0.0, negative = 0.0), length = 4, offset = 43, is_negated = False)])])],
		'is_error': False
	}
]`



**Test using Postman on /text/analytics/v3.1/sentiment?opinionMining=true
Results (includes Opinion Mining)**
```
{
    "documents": [
        {
            "id": "1",
            "sentiment": "positive",
            "confidenceScores": {
                "positive": 1.0,
                "neutral": 0.0,
                "negative": 0.0
            },
            "sentences": [
                {
                    "sentiment": "positive",
                    "confidenceScores": {
                        "positive": 1.0,
                        "neutral": 0.0,
                        "negative": 0.0
                    },
                    "offset": 0,
                    "length": 13,
                    "text": "Food is great",
                    "targets": [
                        {
                            "sentiment": "positive",
                            "confidenceScores": {
                                "positive": 1.0,
                                "negative": 0.0
                            },
                            "offset": 0,
                            "length": 4,
                            "text": "Food",
                            "relations": [
                                {
                                    "relationType": "assessment",
                                    "ref": "#/documents/0/sentences/0/assessments/0"
                                }
                            ]
                        }
                    ],
                    "assessments": [
                        {
                            "sentiment": "positive",
                            "confidenceScores": {
                                "positive": 1.0,
                                "negative": 0.0
                            },
                            "offset": 8,
                            "length": 5,
                            "text": "great",
                            "isNegated": false
                        }
                    ]
                }
            ],
            "warnings": []
        },
        {
            "id": "2",
            "sentiment": "negative",
            "confidenceScores": {
                "positive": 0.29,
                "neutral": 0.01,
                "negative": 0.7
            },
            "sentences": [
                {
                    "sentiment": "negative",
                    "confidenceScores": {
                        "positive": 0.29,
                        "neutral": 0.01,
                        "negative": 0.7
                    },
                    "offset": 0,
                    "length": 56,
                    "text": "Service is poor but the restaurant is at a good location",
                    "targets": [
                        {
                            "sentiment": "negative",
                            "confidenceScores": {
                                "positive": 0.01,
                                "negative": 0.99
                            },
                            "offset": 0,
                            "length": 7,
                            "text": "Service",
                            "relations": [
                                {
                                    "relationType": "assessment",
                                    "ref": "#/documents/1/sentences/0/assessments/0"
                                }
                            ]
                        },
                        {
                            "sentiment": "positive",
                            "confidenceScores": {
                                "positive": 1.0,
                                "negative": 0.0
                            },
                            "offset": 48,
                            "length": 8,
                            "text": "location",
                            "relations": [
                                {
                                    "relationType": "assessment",
                                    "ref": "#/documents/1/sentences/0/assessments/1"
                                }
                            ]
                        }
                    ],
                    "assessments": [
                        {
                            "sentiment": "negative",
                            "confidenceScores": {
                                "positive": 0.01,
                                "negative": 0.99
                            },
                            "offset": 11,
                            "length": 4,
                            "text": "poor",
                            "isNegated": false
                        },
                        {
                            "sentiment": "positive",
                            "confidenceScores": {
                                "positive": 1.0,
                                "negative": 0.0
                            },
                            "offset": 43,
                            "length": 4,
                            "text": "good",
                            "isNegated": false
                        }
                    ]
                }
            ],
            "warnings": []
        }
    ],
    "errors": [],
    "modelVersion": "2022-06-01"
}
```



**Test using Postman on /language/:analyze-text?api-version=2022-05-01
with "opinionMining":true in "parameters"
Results (includes Opinion Mining)**

```
{
    "kind": "SentimentAnalysisResults",
    "results": {
        "documents": [
            {
                "id": "1",
                "sentiment": "positive",
                "confidenceScores": {
                    "positive": 0.99,
                    "neutral": 0.01,
                    "negative": 0.0
                },
                "sentences": [
                    {
                        "sentiment": "positive",
                        "confidenceScores": {
                            "positive": 1.0,
                            "neutral": 0.0,
                            "negative": 0.0
                        },
                        "offset": 0,
                        "length": 46,
                        "text": "The staff are always so friendly and helpful. ",
                        "targets": [
                            {
                                "sentiment": "positive",
                                "confidenceScores": {
                                    "positive": 1.0,
                                    "negative": 0.0
                                },
                                "offset": 4,
                                "length": 5,
                                "text": "staff",
                                "relations": [
                                    {
                                        "relationType": "assessment",
                                        "ref": "#/documents/0/sentences/0/assessments/0"
                                    },
                                    {
                                        "relationType": "assessment",
                                        "ref": "#/documents/0/sentences/0/assessments/1"
                                    }
                                ]
                            }
                        ],
                        "assessments": [
                            {
                                "sentiment": "positive",
                                "confidenceScores": {
                                    "positive": 1.0,
                                    "negative": 0.0
                                },
                                "offset": 24,
                                "length": 8,
                                "text": "friendly",
                                "isNegated": false
                            },
                            {
                                "sentiment": "positive",
                                "confidenceScores": {
                                    "positive": 1.0,
                                    "negative": 0.0
                                },
                                "offset": 37,
                                "length": 7,
                                "text": "helpful",
                                "isNegated": false
                            }
                        ]
                    },
                    {
                        "sentiment": "positive",
                        "confidenceScores": {
                            "positive": 0.99,
                            "neutral": 0.01,
                            "negative": 0.0
                        },
                        "offset": 46,
                        "length": 37,
                        "text": "They have to be the best staff around",
                        "targets": [
                            {
                                "sentiment": "positive",
                                "confidenceScores": {
                                    "positive": 0.99,
                                    "negative": 0.01
                                },
                                "offset": 71,
                                "length": 5,
                                "text": "staff",
                                "relations": [
                                    {
                                        "relationType": "assessment",
                                        "ref": "#/documents/0/sentences/1/assessments/0"
                                    }
                                ]
                            }
                        ],
                        "assessments": [
                            {
                                "sentiment": "positive",
                                "confidenceScores": {
                                    "positive": 0.99,
                                    "negative": 0.01
                                },
                                "offset": 66,
                                "length": 4,
                                "text": "best",
                                "isNegated": false
                            }
                        ]
                    }
                ],
                "warnings": []
            },
            {
                "id": "2",
                "sentiment": "positive",
                "confidenceScores": {
                    "positive": 0.97,
                    "neutral": 0.02,
                    "negative": 0.0
                },
                "sentences": [
                    {
                        "sentiment": "positive",
                        "confidenceScores": {
                            "positive": 0.97,
                            "neutral": 0.02,
                            "negative": 0.0
                        },
                        "offset": 0,
                        "length": 62,
                        "text": "Annette at Miranda provided an exceptional customer experience",
                        "targets": [
                            {
                                "sentiment": "positive",
                                "confidenceScores": {
                                    "positive": 1.0,
                                    "negative": 0.0
                                },
                                "offset": 43,
                                "length": 19,
                                "text": "customer experience",
                                "relations": [
                                    {
                                        "relationType": "assessment",
                                        "ref": "#/documents/1/sentences/0/assessments/0"
                                    }
                                ]
                            },
                            {
                                "sentiment": "positive",
                                "confidenceScores": {
                                    "positive": 1.0,
                                    "negative": 0.0
                                },
                                "offset": 0,
                                "length": 18,
                                "text": "Annette at Miranda",
                                "relations": [
                                    {
                                        "relationType": "assessment",
                                        "ref": "#/documents/1/sentences/0/assessments/0"
                                    }
                                ]
                            }
                        ],
                        "assessments": [
                            {
                                "sentiment": "positive",
                                "confidenceScores": {
                                    "positive": 1.0,
                                    "negative": 0.0
                                },
                                "offset": 31,
                                "length": 11,
                                "text": "exceptional",
                                "isNegated": false
                            }
                        ]
                    }
                ],
                "warnings": []
            },
            {
                "id": "3",
                "sentiment": "positive",
                "confidenceScores": {
                    "positive": 0.99,
                    "neutral": 0.0,
                    "negative": 0.0
                },
                "sentences": [
                    {
                        "sentiment": "positive",
                        "confidenceScores": {
                            "positive": 0.99,
                            "neutral": 0.0,
                            "negative": 0.0
                        },
                        "offset": 0,
                        "length": 56,
                        "text": "Expert advice & being remembered by the sales assistant.",
                        "targets": [],
                        "assessments": []
                    }
                ],
                "warnings": []
            }
        ],
        "errors": [],
        "modelVersion": "2022-06-01"
    }
}
```

**Comment by mhamilton723** [comment]
Thank you! this is a known issue and it should be fixed in v0.10.0

**Comment by kenshinee** [comment]
@mhamilton723 thank you and much appreciated.

I see that v0.10.0 was released a few days ago. On Azure Synapse Analytics, will it be deployed automatically or we need to upgrade our spark pools from 3.1 to 3.2 to see the changes?

Reference.
![image](https://user-images.githubusercontent.com/863148/180648493-acdf30b5-58eb-4f76-9abf-b1fa3ebd559a.png)

**Comment by serena-ruan** [comment]
You need to create a spark 3.2 pool and add above config in the first cell of your code to install SynapseML v0.10.0. We will add it into Synapse Analytics as default at some point, but at this time you need to install it yourself 😃

**Comment by kenshinee** [comment]
Tested on v0.10.0. Opinion mining available on Synapse Spark 3.2 using the config method listed avove. Closing this now.
